drv/lpc55-update-server/src/main.rs

```diff
@@ -110,34 +112,27 @@ impl idl::InOrderUpdateImpl for ServerImpl {
             return Err(UpdateError::BadLength.into());
         }
 
-        let mut flash_page: [u8; BLOCK_SIZE_BYTES] = [0; BLOCK_SIZE_BYTES];
+        let mut flash_page = [0u8; BLOCK_SIZE_BYTES];
 
-        block
-            .read_range(0..len as usize, &mut flash_page)
-            .map_err(|_| RequestError::Fail(ClientError::WentAway))?;
+        if block_num == HEADER_BLOCK {
+            self.block0 = Some([0u8; BLOCK_SIZE_BYTES]);
+            let block0 = &mut self.block0.as_mut().unwrap_lite()[..];
+            block
+                .read_range(0..len as usize, block0)
+                .map_err(|_| RequestError::Fail(ClientError::WentAway))?;
 
-        flash_page[len..].fill(0);
+            block0[len..].fill(0);
```

Comment:
Now we should do a sanity check on the contents.
The correct MAGIC should be there.
There should be a SKU and bank indication.
The vector table and entry point should be within the range of blocks being written.

Reply:
Doing the sanity check in the prepare call would leave the bank intact; an argument for including the header as a parameter to the prepare call.
If the prepare call does no bank erase, and the header containing block is the first we see, then we're still ok.

Reply:
Do we have a SKU? 

I can add the contents sanity check to look for the MAGIC and other things.  I don't want to change the API in this PR, because there are other PRs in flight that use it and I'd prefer not to have this interfere. I agree we should revisit though.

Reply:
Agreed.

Reply:
I added a check for header magic and that the image for the right slot is being sent via an entry-point check in the header.